### PR TITLE
Refactor the module code-base

### PR DESCRIPTION
**Pull request by ayeshLK**

### CODE REVIEW

File: websub-ballerina/commons.bal [hunk 1]
```diff
@@ -145,253 +83,13 @@ public type Acknowledgement record {
...
-
-    if (hubs.length() > 0 && topic != "") {
-        return [topic, hubs];
-    }
-    return error WebSubError("Hub and/or Topic URL(s) not identified in link header of discovery response");
-}
+public type RequestQueryParams record {|
+    string hubMode = "";
+    string hubTopic = "";
+    string hubChallenge = "";
```
Comment (shafreenAnfar) [inline websub-ballerina/commons.bal:89]: Above doesn't look like ballerina way of coding :) Any reason why we don't make those `string?` ?

File: websub-ballerina/http_service.bal [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                "onSubscriptionValidationDenied" => {
+                    self.isSubscriptionValidationDeniedAvailable = true;
+                }
+                "onSubscriptionVerification" => {
+                    self.isSubscriptionVerificationAvailable = true;
+                }
+                "onEventNotification" => {
+                    self.isEventNotificationAvailable = true;
+                }
+                _ => {}
```
Comment (shafreenAnfar) [inline websub-ballerina/http_service.bal:54]: I think it is better log something here. Maybe an error log ?

File: websub-ballerina/request_processor.bal [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            };              
+        }
+        "application/octet-stream" => {
+            message = {
+                headers: headers,
+                contentType: "application/octet-stream",
+                content: checkpanic request.getBinaryPayload()
+            };  
+        }
+        _ => {}
```
Comment (shafreenAnfar) [inline websub-ballerina/request_processor.bal:114]: ditto

File: websub-ballerina/request_processor.bal [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+    if (!isVerifiedContent) {
+        return;
+    }
+                                               
+    string contentType = request.getContentType();
+
+    map<string|string[]> headers = retrieveRequestHeaders(request);
+
+    ContentDistributionMessage? message = ();
```
Comment (shafreenAnfar) [inline websub-ballerina/request_processor.bal:83]: string contentType = request.getContentType();
map<string|string[]> headers = retrieveRequestHeaders(request);
ContentDistributionMessage? message = ();

I think we can keep those close to each other as they are associated to each other. The stronger that affinity, the less vertical distance there should be between them as explained in Clean Code.

File: websub-ballerina/subscriber_service.bal [hunk 5]
```diff
@@ -17,11 +17,12 @@
 # The WebSub service type
 public type SubscriberService service object {
 
-//  remote function onSubscriptionValidationDenied(SubscriptionDeniedError msg) returns Acknowledgement;
+//  remote function onSubscriptionValidationDenied(SubscriptionDeniedError msg) returns Acknowledgement?;
 
 //  remote function onSubscriptionVerification(SubscriptionVerification msg)
-//                    returns SubscriptionVerificationSuccess|SubscriptionVerificationError;
+//                    returns SubscriptionVerificationSuccess | SubscriptionVerificationError;
```
Comment (shafreenAnfar) [inline websub-ballerina/subscriber_service.bal:23]: Let's remove the spacing around the "|".

File: websub-ballerina/utils.bal [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        "sha256" => {
+            hashedContent = crypto:hmacSha256(contentPayload, keyArr);
+        }
+        "sha384" => {
+            hashedContent = crypto:hmacSha384(contentPayload, keyArr);
+        }
+        "sha512" => {
+            hashedContent = crypto:hmacSha512(contentPayload, keyArr);
+        }
+        _ => {}
```
Comment (shafreenAnfar) [inline websub-ballerina/utils.bal:143]: ditto.

File: websub-ballerina/commons.bal (hunk 1 above)
Comment (ayeshLK) [reply websub-ballerina/commons.bal]: * Updated the `RequestQueryParams` type.

File: websub-ballerina/request_processor.bal (hunk 3 above)
Comment (ayeshLK) [reply websub-ballerina/request_processor.bal]: * Updated with necessary logs.

File: websub-ballerina/utils.bal (hunk 6 above)
Comment (ayeshLK) [reply websub-ballerina/utils.bal]: * Updated with necessary logs.

File: websub-ballerina/subscriber_service.bal (hunk 5 above)
Comment (ayeshLK) [reply websub-ballerina/subscriber_service.bal]: * Removed unnecessary white-spaces.

File: websub-ballerina/request_processor.bal (hunk 4 above)
Comment (ayeshLK) [reply websub-ballerina/request_processor.bal:82]: * Updated the structure accordingly.

File: websub-ballerina/http_service.bal (hunk 2 above)
Comment (ayeshLK) [reply websub-ballerina/http_service.bal]: * Updated with necessary logs.

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**